infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -460,18 +461,17 @@ private void destroyMachineQuietly(String machineName, DockerMachine machine) {
   private class AbnormalMachineStopHandlerImpl implements AbnormalMachineStopHandler {
     @Override
     public void handle(String error) {
+      eventService.publish(
+          new RuntimeAbnormalStoppingEvent(
+              getContext().getIdentity(), "Machine is abnormal stopped."));
       try {
         internalStop(emptyMap());
       } catch (InfrastructureException e) {
         LOG.error(e.getLocalizedMessage(), e);
       } finally {
         eventService.publish(
-            DtoFactory.newDto(RuntimeStatusEvent.class)
-                .withIdentity(DtoConverter.asDto(getContext().getIdentity()))
-                .withStatus("STOPPED")
-                .withPrevStatus("RUNNING")
-                .withFailed(true)
-                .withError(error));
+            new RuntimeAbnormalStoppedEvent(
+                getContext().getIdentity(), "Machine is abnormal stopped."));
```

Comment:
The machine is abnormally stopped. ?

Reply:
Changed to propagating an original error
https://github.com/eclipse/che/blob/07263f1e30089689d71b057f747a44a29283e3c4/infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/monit/DockerMachineStopDetector.java#L151